chemprop/models/multi.py

```diff
@@ -46,3 +46,38 @@ def fingerprint(
         H = self.bn(H)
 
         return H if X_f is None else torch.cat((H, X_f), 1)
+
+    @classmethod
+    def load_from_checkpoint(
+        cls, checkpoint_path, map_location=None, hparams_file=None, strict=True, **kwargs
+    ) -> MPNN:
+        hparams = torch.load(checkpoint_path)["hyper_parameters"]
+
+        kwargs = construct_modules(hparams, **kwargs)
+
+        return super().load_from_checkpoint(
+            checkpoint_path, map_location, hparams_file, strict, **kwargs
+        )
+
+
+def construct_modules(hparams: dict, **kwargs):
+    # construct the `agg` and `predictor` modules from their hparam dicts like before
+    # and store them into `kwargs`
+    agg_hparams = hparams["agg"]
+    agg = agg_hparams.pop("cls")(**agg_hparams)
+    kwargs["agg"] = agg
+
+    predictor_hparams = hparams["predictor"]
+    predictor = predictor_hparams.pop("cls")(**predictor_hparams)
+    kwargs["predictor"] = predictor
```

Comment:
do we need these lines? Does loading still work if you comment them out? My thinking is that they should get loaded properly in `super().load_from_checkpoint`

Reply:
Thanks for the suggestion! These are actually not needed. And by removing these, I am able to move the codes inside `construct_modules` back into the `load_from_checkpoint` function.